spec/controllers/api/v2/searches_controller_spec.rb

```diff
@@ -27,9 +27,18 @@
     }
   end
   let(:routed_query_setup) do
-    routed_query = affiliate.routed_queries.build(url: "http://www.gov.gov/foo.html", description: "testing")
+    routed_query = affiliate.routed_queries.build(url: 'http://www.gov.gov/foo.html',
+                                                  description: 'testing')
     routed_query.routed_query_keywords.build(keyword: 'foo bar')
     routed_query.save!
+
+    # Routed queries pass a mocked search object
+    # with the QRTD module
+    # to the Search Impression logger.
+    long_class_name = RoutedQueryImpressionLogger::QueryRoutedSearch
+    mock_search = instance_double(long_class_name, modules: ['QRTD'])
+    allow(long_class_name).to receive(:new).with(['QRTD'], {}).and_return(mock_search)
+    expect(SearchImpression).to receive(:log).with(mock_search, any_args)
```

Comment:
This setup is heavily dependent on the underlying functionality of the `RoutedQueryImpressionLogger` class. I would replace this with simply:
```
    expect(RoutedQueryImpressionLogger).to receive(:log).
      with(affiliate, 'foo bar', an_instance_of(ActionController::TestRequest))
```
...and trust `RoutedQueryImpressionLogger` to do the right thing with those args.

Reply:
Ahhhh, so much cleaner. Thank you. There aren't any tests for `RoutedQueryImpressionLogger` should I include one in this PR? It is also used in the non-api routed query code.

Reply:
Yikes - yes, please!